pkg/plugin/resources.go

```diff
@@ -162,6 +170,21 @@ func (app *App) handleReport(w http.ResponseWriter, req *http.Request) {
 	var credential client.Credential
 
 	switch {
+	// Requires Grafana v10.3.0+ and feature flag "idForwarding" enabled
+	case req.Header.Get(GrafanaUserSignInTokenHeaderName) != "":
+		ctxLogger.Debug("using Grafana user sign-in token")
+
+		credential = client.Credential{
+			HeaderName:  GrafanaUserSignInTokenHeaderName,
+			HeaderValue: req.Header.Get(GrafanaUserSignInTokenHeaderName),
+		}
+	case req.Header.Get(backend.OAuthIdentityTokenHeaderName) != "":
+		ctxLogger.Debug("using OAuth identity token")
+
+		credential = client.Credential{
+			HeaderName:  backend.OAuthIdentityTokenHeaderName,
+			HeaderValue: req.Header.Get(backend.OAuthIdentityTokenHeaderName),
+		}
```

Comment:
Why this case has been moved above `cookies`? AFAIK Grafana does not forward any sort of auth headers to plugins. So this case is ineffective. Even if Grafana does, we should proriotize `idForwarding` tokens and `cookies` as they have correct permissions set based on the user making the requests.

Reply:
Then, we can remove this case?

Reply:
Yes, I guess we can remove it.

Reply:
```suggestion
```